src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -248,6 +261,43 @@ private function autowireMethod($id, Definition $definition, \ReflectionMethod $
     }
 
     /**
+     * Getter injection.
+     *
+     * @param string            $id
+     * @param Definition        $definition
+     * @param \ReflectionMethod $reflectionMethod
+     *
+     * @return bool
+     */
+    private function overrideGetter($id, Definition $definition, \ReflectionMethod $reflectionMethod)
+    {
+        if (!method_exists($reflectionMethod, 'getReturnType')) {
+            return false;
+        }
+
+        if (0 !== $reflectionMethod->getNumberOfParameters() || $reflectionMethod->isFinal() || $reflectionMethod->returnsReference() || !($returnType = $reflectionMethod->getReturnType())) {
+            return false;
+        }
+
+        $class = (string) $returnType;
```

Comment:
In fact, `ReflectionType::__toString` is deprecated since PHP 7.1, you should use this instead:
on https://github.com/symfony/symfony/blob/master/src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php#L1786